### PR TITLE
FE-967 - Hibana style cleanup for sending domains

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-967](https://jira.int.messagesystems.com/browse/FE-967)

### What Changed
- Exposes `marginTop` prop on the `<ButtonWrapper/>` component. Essentially used to remove the margin when these components exist within `<Panel.Section/>` components
- Updates `<ButtonWrapper/>` component documentation accordingly
- Updates `<ButtonWrapper/>` imports to be `import { ButtonWrapper } from 'src/components';`
- Updates destructive button variant to match Patrick's implementation in FE-962
- Adds new `destructive-secondary` variant to match Patrick's implementation in FE-962
- Updates `<ConfirmationModal/>` to match Patrick's implementation in FE-962, allowing passing in button variants
- Adds new `<SubduedText/>` global component that renders as light-ish gray text in Hibana and a regular paragraph in the OG version of the app

### How To Test
- Verify Sending Domains pages are updated to be (fairly) consistent with the mocks

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

File: src/pages/sendingDomains/components/SetupInstructionPanel.js [hunk 1]
```diff
@@ -10,30 +12,33 @@ const SetupInstructionPanel = ({
...
+          content: isVerified ? `Re-verify ${recordType} Record` : `Verify ${recordType} Record`,
+          disabled: isVerifying,
+          id: verifyButtonIdentifier,
+          onClick: onVerify,
+        },
+      ].filter(Boolean)}
+      title={
+        <div className={styles.Title}>
+          <span>DNS Settings</span>
+          <span className={styles.TitleIcon}>{isVerified ? <VerifiedIcon /> : <ErrorIcon />}</span>
```
Comment (nicklemmon) [inline src/pages/sendingDomains/components/SetupInstructionPanel.js:33]: Added some new class names to fix a bug - vertical centering was pretty badly off

File: src/pages/sendingDomains/components/StatusLabel.js [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 import { DomainStatusTag } from 'src/components/tags';
 import { VerifiedIcon } from './Icons';
 
 const StatusLabel = ({ status }) => {
   if (status === 'verified') {
-    return <div><VerifiedIcon/> <strong>Verified</strong></div>;
+    return (
+      <Box display="flex" alignItems="center">
+        <VerifiedIcon /> <strong>Verified</strong>
+      </Box>
```
Comment (nicklemmon) [inline src/pages/sendingDomains/components/StatusLabel.js:11]: Helps with vertical centering bug

File: src/pages/sendingDomains/components/tests/SendingDomainSection.test.js [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
+      children: ['Bill', 'Brasky'],
     };
     const wrapper = shallow(<SendingDomainSection.Left {...props} />);
     expect(wrapper).toMatchSnapshot();
   });
 
   it('should render Right section', () => {
     const props = {
-      children: ['Bill', 'Brasky']
+      children: ['Bill', 'Brasky'],
```
Comment (nicklemmon) [inline src/pages/sendingDomains/components/tests/SendingDomainSection.test.js:31]: I just noticed this - too funny:
https://www.nbc.com/saturday-night-live/video/braskys-buddies-at-the-bar/n11181

File: src/pages/sendingDomains/components/EditBounce.module.scss [hunk 4]
```diff
@@ -1,9 +1,9 @@
 @import '~@sparkpost/matchbox/src/styles/config.scss';
 
 .GridSpacer {
```
Comment (nicklemmon) [inline src/pages/sendingDomains/components/EditBounce.module.scss:3]: Just Prettier - these were intended by 4 spaces for some reason

File: src/components/heading/Heading.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
```
Comment (shrirupa) [inline src/components/heading/Heading.js:1]: 🎉

File: src/pages/sendingDomains/components/SendingDomainSection.js [hunk 6]
```diff
@@ -1,34 +1,52 @@
...
+
+const Section = ({ title, children }) => {
+  const styles = useHibanaOverride(OGStyles, hibanaStyles);
+
+  return (
+    <div
+      className={styles.SendingDomainSection}
+      data-id={`sending-domain-${title.toLowerCase().replace(/\s/g, '-')}`}
+    >
+      <hr className={styles.Hr} />
```
Comment (shrirupa) [inline src/pages/sendingDomains/components/SendingDomainSection.js:36]: Screen Shot 2020-04-27 at 5 12 15 PM
May be we should get rid of these <hr/>, in hibana, as per mocks.

File: src/pages/sendingDomains/components/SendingDomainSection.js [hunk 7]
```diff
@@ -1,34 +1,52 @@
...
+  const styles = useHibanaOverride(OGStyles, hibanaStyles);
+
+  return (
+    <Grid.Column xs={12} lg={5}>
+      <div className={styles.Left}>{children}</div>
+    </Grid.Column>
+  );
+};
+
+const Right = ({ children }) => {
```
Comment (shrirupa) [inline src/pages/sendingDomains/components/SendingDomainSection.js:18]: I remember seeing Left, Right and Section components in abTesting too. May be these should be moved to common components. But I understand if you don't to this with this PR.

Comment (nicklemmon) [reply src/pages/sendingDomains/components/SendingDomainSection.js:18]: They really should be - the way they are handled from a CSS perspective is kind of hacky as they are now too. I'm thinking once we start re-working features we should do that.

**Comment by nicklemmon** [comment]
@shrirupa Ready for another pass! Thanks.

**Pull request merged**